src/components/Nav/useNav.ts

```diff
@@ -465,13 +462,27 @@ export const useNav = () => {
     },
   }
 
+  // Listen for changes to systemTheme and update theme accordingly
+  // Important if the user has not engaged the color mode toggle yet, and
+  // toggles system color preferences
+  useEffect(() => {
+    setTheme("system")
+    setColorMode(systemTheme)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [systemTheme])
+
   const toggleColorMode = () => {
-    setTheme(theme === "dark" ? "light" : "dark")
-    setColorMode(theme === "dark" ? "light" : "dark")
+    // resolvedTheme: "light" | "dark" = Current resolved color mode from useTheme
+    const targetTheme = resolvedTheme === "dark" ? "light" : "dark"
+    // If target theme matches the users system pref, set ls theme to "system"
+    const lsTheme = targetTheme === systemTheme ? "system" : targetTheme
+
+    setTheme(lsTheme)
```

Comment:
When observing this in the Application dev tools, the "system" is automatically parsed (by the embedded `next-theme` I believe*) to "light" or "dark". Using this approach seems to make sure the `html[class]` is updated by the NextThemeProvider.

<details><summary>*embedded `next-theme` script</summary>
<p>

```js
!(function () {
  try {
    var d = document.documentElement,
      c = d.classList;
    c.remove('light', 'dark');
    var e = localStorage.getItem('theme');
    if ('system' === e || (!e && true)) {
      var t = '(prefers-color-scheme: dark)',
        m = window.matchMedia(t);
      if (m.media !== t || m.matches) {
        d.style.colorScheme = 'dark';
        c.add('dark');
      } else {
        d.style.colorScheme = 'light';
        c.add('light');
      }
    } else if (e) {
      c.add(e || '');
    }
    if (e === 'light' || e === 'dark') d.style.colorScheme = e;
  } catch (e) {}
})();
```

</p>
</details>